src/it/scala/net/snowflake/spark/snowflake/IntegrationEnv.scala

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright 2015-2016 Snowflake Computing
+ * Copyright 2015 Databricks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.snowflake.spark.snowflake
+
+import java.sql.{Connection, Timestamp}
+import java.time.{LocalDateTime, ZonedDateTime}
+import java.util.TimeZone
+
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
+import net.snowflake.spark.snowflake.Parameters.MergedParameters
+import org.apache.spark.sql._
+import org.apache.spark.{SparkConf, SparkContext}
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach, FunSuite}
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable
+import scala.io.Source
+
+/**
+  * Base class for writing integration tests which run against a real Snowflake cluster.
+  */
+trait IntegrationEnv
```

Comment:
1. I guess your intention is to reorg "IntegrationSuiteBase" to extract some common code for existing IT test and the new ID test. It is good idea. Could you confirm you did extra change in the same time?

Below is the original traits for IntegrationSuiteBase. I see "FunSuite" is new. Is it intended?

trait IntegrationSuiteBase
    extends QueryTest
    with Matchers
    with BeforeAndAfterAll	
    with BeforeAndAfterEach {

Reply:
Yeah, it's needed for `BeforeAndAfterAll` and `BeforeAndAfterEach`; it was otherwise provided originally by `QueryTest`.